### ISSUE TITLE
Support other HTTP methods rather than just GET in HTTPRequest

### ISSUE THREAD

**Issue by mownier** [godotengine/godot#4028]
**Issue description**
I don't know if it is intended that only the `GET` method to be used in `HTTPRequest`, but it would be great to support `POST`, `PUT`, etc. Currently, the `request` function is:

``` gdscript
request(url, custom_headers, ssl_validate_domain)
```

We could put `method` and `request_data` parameters there with default values `GET` and empty string, respectively:

``` gdscript
request(url, custom_headers, ssl_validate_domain, method, request_data)
```

**Comment by bojidar-bg** [comment]
:+1:

**Comment by neikeq** [comment]
@akien-mga Oops... Github trolled me :P

**Comment by Kazuo256** [comment]
Hi there. I'm interested in trying to make a pull request for this. I've looked through the code of HTTPRequest (in `http_request.cpp`) and found this line:
```c++
Error err = client->request(HTTPClient::METHOD_GET,request_string,headers);
```
Is it enough to change this part? Are there any pitifalls I should be aware of here?

**Comment by reduz** [comment]
I guess you could make it optional

**Comment by Kazuo256** [comment]
Make the "HTTP method type" argument optional, with the default being "GET", right?

OK, I'l try to pull something together.

**Comment by Kazuo256** [comment]
Shouldn't this be closed?

**Comment by bojidar-bg** [comment]
Right... just, next time, instead of writing:

> Implements feature proposed in #4028.

Write:

> ... Closes #4028 ...

So that GH would autoclose it

**Comment by Kazuo256** [comment]
Oh, sorry about that >.<
